R/mod_packageDependencies.R

```diff
@@ -0,0 +1,323 @@
+    }, ignoreInit = TRUE)
+    
+    pkg_df <- eventReactive({selected_pkg$name(); tabready()}, {
+      req(!rlang::is_empty(selected_pkg$name()))
+      req(selected_pkg$name() != "-")
+      req(tabready() == 1L)
+
+      pkginfo <- riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() %>%  
+        as_tibble() %>% 
+        mutate(package = stringr::str_replace(package, "\n", "")) %>% 
+        mutate(name = stringr::word(.data$package, 1, sep = stringr::regex("[\\s|\\(]"))) 
+      
+      # drop any Base R packages from list, unless we ony have 1 row
+      pkginf2 <- pkginfo %>% 
+        filter(!name %in% c(rownames(installed.packages(priority="base")))) 
+      if (nrow(pkginf2) > 0) pkginfo <- pkginf2
+      
+      pkg_name <- ""
+      cl_id <- cli::cli_progress_bar("Assessing Package: ", 
+                                     type = "iterator",
+                                     format = "{pkg_name} {cli::pb_percent} | ETA: {cli::pb_eta}",
+                                     total = nrow(pkginfo))
+      
+      purrr::map_df(pkginfo$name, ~get_versnScore(.x, cl_id), .progress = TRUE) %>% 
+        right_join(pkginfo, by = "name") %>% 
+        select(package, type, name, version, score)
+      
+    }, ignoreInit = TRUE)
+    
+    observeEvent(pkg_df(), {
+      cli::cli_progress_done(id = m_id())
+      cli::cli_progress_done()
+    })
+    
+    # Render Output UI for Package Dependencies.
+    output$package_dependencies_ui <- renderUI({
+      
+      # Lets the user know that a package needs to be selected.
+      if(identical(selected_pkg$name(), character(0)))
+        showHelperMessage()
+      
+      else {
+        fluidPage(
+          
+          shiny::
+            tagList(
+              br(),
+              h4(glue::glue("Package Dependencies: {nrow(depends())}"), style = "text-align: left;"),
+              br(), 
+              tags$strong(glue::glue("First-order dependencies for package: ", {selected_pkg$name()})),
+              br(),
+              fluidRow(
+                column(width = 4,
+                       DT::renderDataTable(server = FALSE, {
+                         
+                         my_data_table <- reactive({
+                           cbind(pkg_df(), 
+                                 data.frame(
+                                   Actions = shinyInput(actionButton, nrow(pkg_df()),
+                                                        'button_',
+                                                        size = "xs",
+                                                        style='height:24px; padding-top:1px;',
+                                                        label = icon("arrow-right", class="fa-regular", lib = "font-awesome"),
+                                                        onclick = paste0('Shiny.onInputChange(\"' , ns("select_button"), '\", this.id)')
```

Comment:
A couple things on this:

1. `Shiny.onInputChange` is the older version of `Shiny.setInputValue`.
2. This needs to to `Shiny.setInputValue` so that the event can be triggered if the same option is clicked.

For example, I loaded `tidyverse` and checked out `dplyr`. I then selected `tidyverse` again to check out `dplyr` again, but nothing happens because of lazy reactivity. For more information, you can read about it [here](https://shiny.posit.co/r/articles/build/communicating-with-js/).

Reply:
OK I changed it to Shiny.setInputValue